app/src/main/java/com/rkm/tasky/database/dao/AgendaDao.kt

```diff
@@ -0,0 +1,95 @@
+package com.rkm.tasky.database.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import androidx.room.Transaction
+import com.rkm.tasky.database.TaskyDatabase
+import com.rkm.tasky.database.model.AgendaDetails
+import com.rkm.tasky.database.model.AttendeeEntity
+import com.rkm.tasky.database.model.EventAttendeesAndPhotosToDelete
+import com.rkm.tasky.database.model.EventEntity
+import com.rkm.tasky.database.model.EventWithDetails
+import com.rkm.tasky.database.model.PhotoEntity
+import com.rkm.tasky.database.model.ReminderEntity
+import com.rkm.tasky.database.model.TaskEntity
+
+@Dao
+interface AgendaDao {
+
+    @Transaction
+    @Query(
+        """
+    SELECT * FROM events WHERE  
+    (
+        ( (`from` >= :startTime) AND (`from` < :endTime) )
+        OR
+        ( (`to` > :startTime) AND (`to` <= :endTime) )
+        OR
+        ( (`from` <= :startTime) AND (`to` > :endTime) )
+    )
+        """
+    )
+    suspend fun getEventsInTimeRange(startTime: Long, endTime: Long): List<EventWithDetails>
```

Comment:
This query can be simplified. The criteria are:
- It starts or ends within the specified dates,
- It completely covers the entire date range.

And the same can be done with this query:
```
SELECT * FROM events 
WHERE `from` < :endTime AND `to` > :startTime
```